packages/run-protocol/src/vpool-xyk-amm/singlePool.js

```diff
@@ -1,104 +0,0 @@
-// @ts-check
-
-import { Far } from '@endo/marshal';
-import { makeFeeRatio } from './constantProduct/calcFees.js';
-import {
-  pricesForStatedInput,
-  pricesForStatedOutput,
-} from './constantProduct/calcSwapPrices.js';
-
-/**
- * @param {ContractFacet} zcf
- * @param {XYKPool} pool
- * @param {() => bigint} getProtocolFeeBP - retrieve governed protocol fee value
- * @param {() => bigint} getPoolFeeBP - retrieve governed pool fee value
- * @param {ZCFSeat} feeSeat
- * @param {AddLiquidityActual} addLiquidityActual
- * @returns {VPoolWrapper<SinglePoolInternalFacet>}
- */
-export const makeSinglePool = (
-  zcf,
-  pool,
-  getProtocolFeeBP,
-  getPoolFeeBP,
-  feeSeat,
-  addLiquidityActual,
-) => {
```

Comment:
Why can you get rid of singlePool, but not doublePool? They wrap pool.js in similar ways.

Reply:
There is a cycle between pool and singlePool (aka vPool). Creating a pool creates a vPool, and they point at each other. Following dependencies starting from `pool.js` I didn't encounter doublePool and so have not started thinking about how to virtualize it. What would you recommend?

Reply:
Not sure.

`doublePool` and `singlePool` are polymorphic wrappers of `pool`. Each pool has a single long-lived `singlePool`. `doublePools` connect two `pool`s, and are created on demand.

I don't think it works to move `singlePool's methods into `pool`.

I don't think it's necessary for `singlePool` to be part of the persistent state. It would be fine for the `singlePool` to be created when the `pool` is created or revived.

Reply:
Just to clarify, can anyone outside the vat be holding a remote reference to a `singlePool`? If so, are `singlePool`s potentially high cardinality?

Reply:
Unlike vaults, clients don't hold onto any state of the AMM between transactions. All requests go through the public (or creator) interface for every request.

Reply:
Excellent! That yes, I can and should simplify this. Thanks.